source/MaterialXFormat/XmlExport.cpp

```diff
@@ -39,46 +41,64 @@ XmlExportOptions::XmlExportOptions() :
 
 void exportToXmlStream(DocumentPtr doc, std::ostream& stream, const XmlExportOptions* exportOptions)
 {
-    mergeLooks(doc, exportOptions);
+    DocumentPtr exportDoc = doc;
+    if (exportOptions && !exportOptions->modifyInPlace)
+    {
+        exportDoc = doc->copy();
+    }
+
+    mergeLooks(exportDoc, exportOptions);
     if (exportOptions && exportOptions->flattenFilenames)
     {
-        flattenFilenames(doc, exportOptions->resolvedTexturePath, exportOptions->stringResolver);
+        flattenFilenames(exportDoc, exportOptions->resolvedTexturePath, exportOptions->stringResolver);
```

Comment:
Probably wasn't you, but what happened to the export resolver resolve loop that exists in the other two export methods?